src/openMVG/sfm/pipelines/sequential/sequential_SfM.cpp

```diff
@@ -882,71 +1039,125 @@ bool SequentialSfMReconstructionEngine::FindImagesWithPossibleResection(
+  std::cout << std::endl;
 
   // If the list is empty or if the list contains images with no correspondences
   // -> (no resection will be possible)
-  if (vec_putative.empty() || vec_putative[0].second == 0)
+  if (vec_putative.empty() || std::get<1>(vec_putative[0]) < minPointsThreshold)
   {
+    std::cout << "FindImagesWithPossibleResection failed: ";
+    if(vec_putative.empty())
+    {
+      std::cout << "No putative image.";
+    }
+    else
+    {
+      std::cout << "Not enough point in the putative images: ";
+      for(auto v: vec_putative)
+        std::cout << std::get<1>(v) << ", ";
+    }
+    std::cout << std::endl;
     // All remaining images cannot be used for pose estimation
     set_remainingViewId.clear();
     return false;
   }
 
-  // Add the image view index that share the most of 2D-3D correspondences
-  vec_possible_indexes.push_back(vec_putative[0].first);
+  // Add the image view index with the best score
+  vec_possible_indexes.push_back(std::get<0>(vec_putative[0]));
 
-  // Then, add all the image view indexes that have at least N% of the number of the matches of the best image.
-  const IndexT M = vec_putative[0].second; // Number of 2D-3D correspondences
-  const size_t threshold = static_cast<size_t>(dThresholdGroup * M);
-  for (size_t i = 1; i < vec_putative.size() &&
-    vec_putative[i].second > threshold; ++i)
+  if (_sfm_data.GetPoses().size() < 30)
   {
-    vec_possible_indexes.push_back(vec_putative[i].first);
+    // Add images one by one to reconstruct the first cameras.
+    std::cout << "FindImagesWithPossibleResection with few images. " << " images took: " << std::chrono::duration_cast<std::chrono::milliseconds>(std::chrono::steady_clock::now() - chrono_start).count() << " msec" << std::endl
+              << " - Scores: " << std::get<2>(vec_putative.front()) << std::endl
+              << " - Features: " << std::get<1>(vec_putative.front()) << std::endl;
+    return true;
   }
+  // Then, add all the image view indexes that have at least N% of the score of the best image.
+#ifdef OPENMVG_NEXTBESTVIEW_WITHOUT_SCORE
+  const IndexT M = std::get<2>(vec_putative[0]); // View score based on the number of 2D-3D correspondences and their repartition.
+  const size_t scoreThreshold = dThresholdGroup * M;
+#else
+  const size_t scoreThreshold = _pyramidThreshold;
+#endif
+  for (size_t i = 1;
+       i < 30 && // max of 30 images per group
+       i < vec_putative.size() &&
+       std::get<1>(vec_putative[i]) > minPointsThreshold && // ensure min number of points
+       std::get<2>(vec_putative[i]) > scoreThreshold; // ensure score level
+       ++i)
+  {
+    vec_possible_indexes.push_back(std::get<0>(vec_putative[i]));
+    if(!std::get<3>(vec_putative[i]))
+    {
+      // If we add a new intrinsic, it is a sensitive stage in the process,
+      // so it is better to perform a Bundle Adjustment just after.
+      break;
```

Comment:
IMHO it would be better return all the results and let the callee decide what to do, it can check if the intrinsics are unknown and run a BA. I think it's better for code reusability as it may be useful for another part of the reconstruction pipeline or simply use a different policy for new intrinsics.